app/models/monthly_statistics/by_status.rb

```diff
@@ -0,0 +1,161 @@
+        tally
+      end
+
+      combined_tally = status_for_choices_tally + format_deferral_status_choices_tally
+
+      status_and_count_hash = combined_tally.map do |hash|
+        { hash['status'] => hash['count'] }
+      end
+
+      status_and_count_hash.each_with_object(Hash.new(0)) do |hash, sum|
+        hash.each { |key, value| sum[key] += value }
+      end
+    end
+
+    def tally_application_choices(phase:, cycle:, field:)
+      without_subsequent_applications_query = if @by_candidate
+                                                "AND (
+                                                  NOT EXISTS (
+                                                    SELECT 1
+                                                    FROM application_forms
+                                                    AS subsequent_application_forms
+                                                    WHERE application_forms.id = subsequent_application_forms.previous_application_form_id
+                                                  )
+                                                )"
+                                              else
+                                                ''
+                                              end
+
+      query = "SELECT COUNT(application_choices_with_minimum_statuses.id), application_choices_with_minimum_statuses.#{field}
+                FROM (
+                  SELECT application_choices.id as id,
+                         application_choices.status_before_deferral as status_before_deferral,
+                         application_choices.status as status,
+                         ROW_NUMBER() OVER (
+                          PARTITION BY application_forms.id
+                          ORDER BY
+                          CASE application_choices.#{field}
+                          WHEN 'recruited' THEN 1
+                          WHEN 'pending_conditions' THEN 2
+                          WHEN 'conditions_not_met' THEN 2
+                          WHEN 'offer' THEN 3
+                          WHEN 'awaiting_provider_decision' THEN 4
+                          WHEN 'interviewing' THEN 4
+                          WHEN 'declined' THEN 5
+                          WHEN 'offer_withdrawn' THEN 6
+                          WHEN 'withdrawn' THEN 7
+                          WHEN 'cancelled' THEN 7
+                          WHEN 'rejected' THEN 7
+                          ELSE 8
+                          END
+                        ) AS row_number
+                        FROM application_forms
+                        INNER JOIN application_choices
+                          ON application_choices.application_form_id = application_forms.id
+                          WHERE application_forms.recruitment_cycle_year = #{cycle}
+                          AND application_forms.phase = '#{phase}'
+                          #{without_subsequent_applications_query}
```

Comment:
Could you handle this by JOINing back onto the application forms table and selecting either `is null` or `is not null` for the joined table, eg sth like (below line 133, and then down in the WHERE)

```sql
INNER JOIN application_forms previous_applications
ON previous_applications.id = application_forms.previous_application_form_id
...
WHERE previous_applications_id IS NULL
```

Reply:
mmm ... not sure I understand entirely sorry. Am I right in thinking that if the above were possible we would still need to switch between `IS NULL` and `IS NOT NULL` depending on the value of `@candidate`?

Reply:
@davidgisbey 

@duncanjbrown and I discussed and agreed to merge. 

There is still some thinking to do about how we make the specs / code more readable so that the business rules are clear / easy to understand but can revisit once this work resumes after End of Cycle ...